cmake/vecmem-check-sycl-source-compiles.cmake

```diff
@@ -0,0 +1,36 @@
+# VecMem project, part of the ACTS project (R&D line)
+#
+# (c) 2024 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+
+# CMake version requirement.
+cmake_minimum_required( VERSION 3.14 )
```

Comment:
One thing I found with these is that they reset policy switches, so my assumption was that it's best practice to avoid them. 

Did you consider this?

Reply:
It indeed resets earlier requirements. But just for the scope of this file. The file including this one should not get its requirements updated. At least I **think** how it's supposed to happen.

This file absolutely needs `3.14` as a minimum. So as long as that minimum doesn't leak into the code including this one, I'm happy with overriding the caller's requirement for just the scope of this file. :thinking: